include/umf/providers/provider_fixed_memory.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_FIXED_MEMORY_PROVIDER_H
+#define UMF_FIXED_MEMORY_PROVIDER_H
+
+#include <umf/providers/provider_os_memory.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @cond
+#define UMF_FIXED_RESULTS_START_FROM 4000
+/// @endcond
+
+struct umf_fixed_memory_provider_params_t;
+
+typedef struct umf_fixed_memory_provider_params_t
+    *umf_fixed_memory_provider_params_handle_t;
+
+/// @brief  Create a struct to store parameters of the Fixed Memory Provider.
+/// @param  hParams [out] handle to the newly created parameters struct.
+/// @param  ptr [in] pointer to the memory region.
+/// @param  size [in] size of the memory region in bytes.
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfFixedMemoryProviderParamsCreate(
+    umf_fixed_memory_provider_params_handle_t *hParams, void *ptr, size_t size);
+
+/// @brief  Destroy parameters struct.
+/// @param  hParams [in] handle to the parameters of the Fixed Memory Provider.
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfFixedMemoryProviderParamsDestroy(
+    umf_fixed_memory_provider_params_handle_t hParams);
```

Comment:
I think we do not need an empty config. Better to accept `NULL` as a params argument in the `umfMemoryProviderCreate`.

Recently for `jemalloc` pool we removed Create/Destroy functions for the config.